cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -118,16 +125,39 @@ public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterv
     return rowKey;
   }
 
+  public byte[] getNextRowKey(List<TagValue> previousTags, byte[] rowKey) {
+    byte[] newRowKey = new byte[rowKey.length];
+    int offset = entityTable.getIdSize() + Bytes.SIZEOF_INT +  entityTable.getIdSize() * previousTags.size();
+    byte[] stopKey = Bytes.stopKeyForPrefix(Arrays.copyOfRange(rowKey,
```

Comment:
Do you assume that given row key already has smth in position of that tagValue to be changed? If so, state that in the javadoc too